src/app/compare/compare.component.ts

```diff
@@ -41,60 +53,59 @@ export class CompareComponent implements OnInit, AfterViewInit {
 
   ngOnInit(): void {
     this.compareData = this.getData(this.getIdsFromPath());
+    this.getStrategyFromLocalStorage();
+  }
+
+  ngAfterViewInit(): void {
     if (this.compareData) {
       this.renderDiffs(this.compareData.originalReport.xml, this.compareData.runResultReport.xml);
+      this.showReports();
     } else {
       this.router.navigate(['debug']);
     }
   }
 
-  ngAfterViewInit(): void {
-    setTimeout(() => {
-      this.showReports();
-    }, 100);
+  private getData(id: string): CompareData | undefined {
+    return this.tabService.activeCompareTabs.get(id);
   }
 
-  getIdsFromPath(): string {
+  private getIdsFromPath(): string {
     return this.route.snapshot.paramMap.get('id') as string;
   }
 
-  getData(id: string): CompareData | undefined {
-    return this.tabService.activeCompareTabs.get(id);
+  private renderDiffs(leftSide: string, rightSide: string): void {
+    this.originalModel = Object.assign({}, this.originalModel, {
+      code: leftSide,
+    });
+    this.modifiedModel = Object.assign({}, this.originalModel, {
+      code: rightSide,
+    });
```

Comment:
This can be changed to use the spread operator https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Reply:
You're right, this was some old code that just got moved, fixed it.